lib/incoming_form.js

```diff
@@ -91,7 +91,10 @@ IncomingForm.prototype.parse = function(req, cb) {
             }
             files[name].push(file);
           } else {
-            files[name] = file;
+            if (!Array.isArray(files[name])) {
```

Comment:
`!Array.isArray(files[name])` is implied by `!files[name]`; this would just be `files[name] = [file];`. (Still, given the `Array.isArray` check above, it looks as if the intent was for single files not to be arrays.)